### PR TITLE
CSIServiceAccountToken ga

### PR DESCRIPTION
**Pull request by zshihang**
#### What type of PR is this?

/kind feature
/kind api-change

xref: https://github.com/kubernetes/enhancements/issues/2047

#### What this PR does / why we need it:
graduate CSIServiceAccountToken feature to stable

#### Does this PR introduce a user-facing change?
```release-note
CSIServiceAccountToken is GA.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1855-csi-driver-service-account-token
```

### CODE REVIEW

File: pkg/registry/storage/csidriver/strategy_test.go [hunk 1]
```diff
@@ -113,21 +113,12 @@ func TestCSIDriverPrepareForCreate(t *testing.T) {
 			name:         "capacity disabled",
 			withCapacity: false,
 		},
-		{
```
Comment (msau42) [inline pkg/registry/storage/csidriver/strategy_test.go:116]: `withServiceAccountToken` definition can be removed.

File: pkg/registry/storage/csidriver/strategy_test.go [hunk 2]
```diff
@@ -167,21 +158,6 @@ func TestCSIDriverPrepareForCreate(t *testing.T) {
 					t.Errorf("VolumeLifecycleModes not stripped: %v", csiDriver.Spec)
 				}
 			}
-			if test.withServiceAccountToken {
-				if csiDriver.Spec.TokenRequests == nil {
```
Comment (msau42) [inline pkg/registry/storage/csidriver/strategy_test.go:171]: Should we keep the assertion that the fields are not dropped?

File: pkg/registry/storage/csidriver/strategy_test.go (hunk 1 above)
Comment (zshihang) [reply pkg/registry/storage/csidriver/strategy_test.go:116]: done

File: pkg/registry/storage/csidriver/strategy_test.go (hunk 2 above)
Comment (zshihang) [reply pkg/registry/storage/csidriver/strategy_test.go:171]: in any case we no longer drop fields so no need to assert fields are not dropped.

Comment (msau42) [reply pkg/registry/storage/csidriver/strategy_test.go:171]: In the base CSIDriver object input to the test, can we set some values to these fields? That way we test that this method no longer mutates those fields.

**Comment by msau42** [comment]
/lgtm
/approve

Just a nit on unit test

File: pkg/registry/storage/csidriver/strategy_test.go (hunk 2 above)
Comment (zshihang) [reply pkg/registry/storage/csidriver/strategy_test.go:171]: we already set. line 133-134.

**Pull request merged**